v2/tools/generator/internal/config/type_configuration.go

```diff
@@ -272,6 +278,18 @@ func (tc *TypeConfiguration) UnmarshalYAML(value *yaml.Node) error {
 			continue
 		}
 
+		// $renameTo: <string>
```

Comment:
Can we document this in the `azure-arm.yaml`? Specifically interested in the "does it only work on resources" versus does it work on all types?

I think it works on all types, but we also have `exportAs`, which I think only works on resources? We should clarify in the documentation when users should use one versus the other and we should make sure that we give a reasonable error if they try to do it wrong.

I'm also interested in what the difference between `exportAs: foo` and `renameTo: foo` + `export: true` are.

Reply:
Docs added.

`$export: true` + `$renameTo: <string>` are very close to `$exportAs` but with the renaming happening at a different pipeline stage, so the results may be subtly different.